File: Scripts/results/generated_code_results/Flutter/bak_code/Productivity/TaskManagerApp/TaskDetailsScreen.dart
import 'package:flutter/material.dart';
import 'dart:math' as math;
class TaskDetailsScreen extends StatelessWidget {
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
leading: IconButton(
icon: Icon(Icons.arrow_back),
onPressed: () => Navigator.pop(context),
),
title: Text('Task Details'),
),
body: SingleChildScrollView(
padding: EdgeInsets.all(16.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(
'User experience design',
style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold),
),
SizedBox(height: 16),
Row(
mainAxisAlignment: MainAxisAlignment.spaceBetween,
children: [
Text('18-02-2022', style: TextStyle(color: Colors.grey)),
Text('9:00 AM-12:00 PM', style: TextStyle(color: Colors.grey)),
],
),
SizedBox(height: 24),
Row(
children: [
Expanded(
flex: 2,
child: CustomPaint(
size: Size(150, 150),
painter: PieChartPainter(),
),
),
Expanded(
flex: 3,
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
_buildLegendItem(Colors.blue, 'Finish on time'),
SizedBox(height: 8),
_buildLegendItem(Colors.orange, 'Past the deadline'),
SizedBox(height: 8),
_buildLegendItem(Colors.purple, 'Still ongoing'),
],
),
),
],
),
SizedBox(height: 24),
Text('Description', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
SizedBox(height: 8),
Text(
"Task management is the process which is monitoring your fast project's tasks through their various stages from start......",
style: TextStyle(color: Colors.grey[600]),
),
TextButton(
onPressed: () {
// Implement see more functionality
},
child: Text('See more'),
),
SizedBox(height: 16),
Text('Sub Task', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
SizedBox(height: 8),
_buildSubTaskItem('Website Redesign', '18 Feb 2022', true),
_buildSubTaskItem('Website Design', '19 Feb 2022', false),
],
),
),
);
}
Widget _buildLegendItem(Color color, String label) {
return Row(
children: [
Container(
width: 16,
height: 16,
color: color,
),
SizedBox(width: 8),
Text(label),
],
);
}
Widget _buildSubTaskItem(String title, String date, bool isCompleted) {
return ListTile(
leading: Checkbox(
value: isCompleted,
onChanged: (bool? value) {
// Implement checkbox functionality
},
),
title: Text(title),
subtitle: Text(date),
trailing: IconButton(
icon: Icon(Icons.more_vert),
onPressed: () {
// Implement more options functionality
},
),
);
}
}
class PieChartPainter extends CustomPainter {
@override
void paint(Canvas canvas, Size size) {
final center = Offset(size.width / 2, size.height / 2);
final radius = math.min(size.width, size.height) / 2;
ini
¸´ÖÆ
final paint = Paint()
  ..style = PaintingStyle.fill;

final total = 100.0;
var startAngle = -math.pi / 2;

void drawSection(double percentage, Color color) {
  final sweepAngle = 2 * math.pi * (percentage / total);
  paint.color = color;
  canvas.drawArc(Rect.fromCircle(center: center, radius: radius),
      startAngle, sweepAngle, true, paint);
  startAngle += sweepAngle;
}

drawSection(40, Colors.blue);
drawSection(40, Colors.orange);
drawSection(20, Colors.purple);
}
@override
bool shouldRepaint(CustomPainter oldDelegate) => false;
}

File: Scripts/results/generated_code_results/RN_gpt/Productivity/TaskManagerApp/TaskDetailsScreen.js
import React from 'react';
import { View, Text, StyleSheet, TouchableOpacity, ImageBackground } from 'react-native';
import { Ionicons } from 'react-native-vector-icons';

const TaskDetailsScreen = ({ navigation }) => {
  return (
    <View style={styles.container}>
      {/* Header with Back Button and Title */}
      <View style={styles.header}>
        <TouchableOpacity onPress={() => navigation.goBack()}>
          <Ionicons name="arrow-back" size={24} color="black" />
        </TouchableOpacity>
        <Text style={styles.headerTitle}>Task Details</Text>
      </View>

      {/* Task Title */}
      <Text style={styles.taskTitle}>User experience design</Text>

      {/* Date and Time */}
      <View style={styles.dateTimeContainer}>
        <View style={styles.dateTimeBox}>
          <Text style={styles.dateTimeText}>18-02-2022</Text>
        </View>
        <View style={styles.dateTimeBox}>
          <Text style={styles.dateTimeText}>9:00 AM-12:00 PM</Text>
        </View>
      </View>

      {/* Pie Chart */}
      <View style={styles.pieChartContainer}>
        <ImageBackground
          source={require('../assets/snack-icon.png')}
          style={styles.pieChart}>
          {/* Placeholder for Pie Chart */}
        </ImageBackground>
        <View style={styles.legend}>
          <Text style={[styles.legendText, { color: 'cyan' }]}>Finish on time</Text>
          <Text style={[styles.legendText, { color: 'orange' }]}>Past the deadline</Text>
          <Text style={[styles.legendText, { color: 'purple' }]}>Still ongoing</Text>
        </View>
      </View>

      {/* Description Section */}
      <Text style={styles.sectionTitle}>Description</Text>
      <Text style={styles.descriptionText}>
        Task management is the process which is monitoring your fast project's tasks through their various stages from start...
        <Text style={styles.seeMore}> See more</Text>
      </Text>

      {/* Sub Task Section */}
      <Text style={styles.sectionTitle}>Sub Task</Text>
      <View style={styles.subTaskContainer}>
        <View style={[styles.subTaskBox, styles.subTaskCompleted]}>
          <TouchableOpacity>
            <Ionicons name="checkmark-circle" size={24} color="cyan" />
          </TouchableOpacity>
          <View style={styles.subTaskContent}>
            <Text style={styles.subTaskTitle}>Website Redesign</Text>
            <Text style={styles.subTaskDate}>18 Feb 2022</Text>
          </View>
          <TouchableOpacity>
            <Ionicons name="ellipsis-horizontal" size={24} color="black" />
          </TouchableOpacity>
        </View>

        <View style={styles.subTaskBox}>
          <TouchableOpacity>
            <Ionicons name="ellipse-outline" size={24} color="gray" />
          </TouchableOpacity>
          <View style={styles.subTaskContent}>
            <Text style={styles.subTaskTitle}>Website Design</Text>
            <Text style={styles.subTaskDate}>19 Feb 2022</Text>
          </View>
          <TouchableOpacity>
            <Ionicons name="ellipsis-horizontal" size={24} color="black" />
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
    backgroundColor: '#fff',
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  headerTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    marginLeft: 10,
  },
  taskTitle: {
    fontSize: 24,
    fontWeight: 'bold',
    marginVertical: 20,
  },
  dateTimeContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 20,
  },
  dateTimeBox: {
    backgroundColor: '#f5f5f5',
    borderRadius: 5,
    padding: 10,
  },
  dateTimeText: {
    fontSize: 16,
  },
  pieChartContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 20,
  },
  pieChart: {
    width: 100,
    height: 100,
  },
  legend: {
    marginLeft: 20,
  },
  legendText: {
    fontSize: 14,
    marginBottom: 5,
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    marginVertical: 10,
  },
  descriptionText: {
    fontSize: 16,
    color: '#555',
    marginBottom: 20,
  },
  seeMore: {
    color: '#007bff',
  },
  subTaskContainer: {
    marginVertical: 10,
  },
  subTaskBox: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#f5f5f5',
    borderRadius: 10,
    padding: 15,
    marginBottom: 10,
  },
  subTaskCompleted: {
    backgroundColor: '#e0f7fa',
  },
  subTaskContent: {
    flex: 1,
    marginLeft: 10,
  },
  subTaskTitle: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  subTaskDate: {
    fontSize: 14,
    color: '#555',
  },
});

export default TaskDetailsScreen;